lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -909,6 +910,28 @@ def test_get_inventory(
             assert file_info["type"] == exp_file_type
             assert file_info["stream"].stream == exp_stream
 
+    def test_tarfile_extract(self, monkeypatch, tmp_path):
+        """Test to check Tarball.extract success"""
+        tar = Path("/mock/result.tar.xz")
+        contents = b"[test]\nfoo=bar\n"
+
+        class MockTarFile:
+            def extractfile(self, path: str) -> IO[bytes]:
+                if path == "metadata.log":
+                    return io.BytesIO(contents)
+                raise Exception("you can't handle exceptions")
+
+        def fake_tarfile_open(tarfile: str, *args):
+            if str(tarfile) == str(tar):
+                return MockTarFile()
+            raise Exception("You didn't see this coming")
```

Comment:
Ha!  And, for both you and Riya, those tests _passed_ even though they weren't producing the desired behavior in the CUT.  😆

In any case, I think it's just a rule of thumb that mocks should not use exceptions to indicate that the mock (or the CUT) is not behaving correctly (because we cannot be sure that that report will be visible to the test code).

But, we don't have to hold the merge for this.  (I just wanted to be sure that you'd made a decision on it.)